src/test/java/com/salesforce/dynamodbv2/QueryTest.java

```diff
@@ -120,149 +138,186 @@ void queryWithKeyConditionsEq(TestArgument testArgument) {
-                .collect(Collectors.toSet()));
-    }
-
-    /**
-     * Table has (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX); we ask for items that match hk and have
-     * rk > RANGE_KEY_N_MIN + 1, so there should be RANGE_KEY_N_MAX - (RANGE_KEY_N_MIN + 1) results:
-     * (hk, RANGE_KEY_N_MIN + 2), ..., (hk, RANGE_KEY_N_MAX).
-     */
     @ParameterizedTest(name = "{arguments}")
-    @ArgumentsSource(DefaultArgumentProviderForTable4.class)
-    void queryWithKeyConditionsGtAlmostAsLow(TestArgument testArgument) {
-        queryWithKeyConditionsComparisonOperatorInner(GT,
-            testArgument,
-            String.valueOf(RANGE_KEY_N_MIN + 1),
-            IntStream.rangeClosed(RANGE_KEY_N_MIN + 2, RANGE_KEY_N_MAX)
-                .mapToObj(String::valueOf)
-                .collect(Collectors.toSet()));
+    @ArgumentsSource(DefaultArgumentProvider.class)
+    @DefaultArgumentProviderConfig(tables = {TABLE4}, useDynalite = true)
+    void queryHkRkWithKeyConditions(TestArgument testArgument) {
+        runHkRkTableKeyConditionTest(testArgument, false);
     }
 
-    /**
-     * Table has (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX); we ask for items that match hk and have
-     * rk >= RANGE_KEY_N_MIN, so there should be RANGE_KEY_N_MAX - RANGE_KEY_N_MIN + 1 results:
-     * (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX).
-     */
     @ParameterizedTest(name = "{arguments}")
-    @ArgumentsSource(DefaultArgumentProviderForTable4.class)
-    void queryWithKeyConditionsGeLow(TestArgument testArgument) {
-        queryWithKeyConditionsComparisonOperatorInner(GE,
-            testArgument,
-            String.valueOf(RANGE_KEY_N_MIN),
-            IntStream.rangeClosed(RANGE_KEY_N_MIN, RANGE_KEY_N_MAX)
-                .mapToObj(String::valueOf)
-                .collect(Collectors.toSet()));
+    @ArgumentsSource(DefaultArgumentProvider.class)
+    @DefaultArgumentProviderConfig(tables = {TABLE4}, useDynalite = true)
+    void queryHkRkWithKeyConditions_legacy(TestArgument testArgument) {
+        runHkRkTableKeyConditionTest(testArgument, true);
     }
 
-    /**
-     * Table has (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX); we ask for items that match hk and have
-     * rk >= RANGE_KEY_N_MIN + 1, so there should be RANGE_KEY_N_MAX - RANGE_KEY_N_MIN results:
-     * (hk, RANGE_KEY_N_MIN + 1), ..., (hk, RANGE_KEY_N_MAX).
-     */
-    @ParameterizedTest(name = "{arguments}")
-    @ArgumentsSource(DefaultArgumentProviderForTable4.class)
-    void queryWithKeyConditionsGeAlmostAsLow(TestArgument testArgument) {
-        queryWithKeyConditionsComparisonOperatorInner(GE,
-            testArgument,
-            String.valueOf(RANGE_KEY_N_MIN + 1),
-            IntStream.rangeClosed(RANGE_KEY_N_MIN + 1, RANGE_KEY_N_MAX)
-                .mapToObj(String::valueOf)
-                .collect(Collectors.toSet()));
+    private void runHkRkTableKeyConditionTest(TestArgument testArgument, boolean useLegacy) {
```

Comment:
We used to have better coverage for queries using legacy key conditions but less coverage for those using key condition expressions, so this fixes that. Also thought about keeping these as different test cases, but with Dynalite already making these tests run slower, and since the meat of test is read-only, here I'm the test cases combined, so we don't have to set up test data once for every case.